src/Microsoft.Tye.Core/ConfigModel/ConfigFactory.cs

```diff
@@ -97,5 +98,8 @@ private static ConfigApplication FromYaml(FileInfo file)
             using var parser = new YamlParser(file);
             return parser.ParseConfigApplication();
         }
+
+        private static string Normalization(string name)
```

Comment:
```suggestion
        private static string NormalizeServiceName(string name)
```

And add a comment about why we are replacing all characters with -. Also a link to the DNS name spec would be useful.

Reply:
2cb2a43, 64a74df these 2 commits aren't mine. 
so just have reverted and pushed. isn't fine? please help me to reslove PR. @davidfowl @jkotalik .

Reply:
I think those commits are fine, but I'd recommend doing a `git rebase -i origin/master`, removing the unnecessary commits, and then doing a `git push -f`. Be careful though!

Reply:
I’ve not done before will do. Thanks  @jkotalik